tests/transformer/test_subscript.py

```diff
@@ -0,0 +1,118 @@
+from tests import e_exec
+
+import pytest
+
+
+GET_SUBSCRIPTS = """
+def simple_subscript(a):
+    return a['b']
+
+def tuple_subscript(a):
+    return a[1, 2]
+
+def slice_subscript_no_upper_bound(a):
+    return a[1:]
+
+def slice_subscript_no_lower_bound(a):
+    return a[:1]
+
+def slice_subscript_no_step(a):
+    return a[1:2]
+
+def slice_subscript_with_step(a):
+    return a[1:2:3]
+
+def extended_slice_subscript(a):
+    return a[0, :1, 1:, 1:2, 1:2:3]
+"""
+
+
+@pytest.mark.parametrize(*e_exec)
+def test_read_subscripts(
+        e_exec, mocker):
+    value = None
+    _getitem_ = mocker.stub()
+    _getitem_.side_effect = lambda ob, index: (ob, index)
+    glb = {'_getitem_': _getitem_}
+    e_exec(GET_SUBSCRIPTS, glb)
+
+    ret = glb['simple_subscript'](value)
+    ref = (value, 'b')
+    assert ref == ret
+    _getitem_.assert_called_once_with(*ref)
+    _getitem_.reset_mock()
```

Comment:
It would be nicer to have separate tests as it would allow to remove the `reset_mock` calls.